### PR TITLE
feat(squads) update posting permission

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- Add permissions to edit squad form
- API implementation here https://github.com/dailydotdev/daily-api/pull/1228

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1196 #done

### CODE REVIEW

File: packages/shared/src/components/modals/EditSquadModal.tsx [hunk 1]
```diff
@@ -46,7 +47,7 @@ function EditSquadModal({
       size={Modal.Size.Small}
       onRequestClose={onRequestClose}
     >
-      <Modal.Header title="Squad details" kind={ModalHeaderKind.Tertiary} />
+      <Modal.Header title="Squad settings" kind={ModalHeaderKind.Primary} />
```
Comment (rebelchris) [inline packages/shared/src/components/modals/EditSquadModal.tsx:50]: Primary is the default so we can omit it here.

File: packages/shared/src/components/squads/Details.tsx [hunk 2]
```diff
@@ -191,24 +191,22 @@ export function SquadDetails({
...
-                name="memberPostingRole"
-                options={memberPostingRoleOptions}
-                value={memberPostingRole}
-                onChange={(value) =>
-                  setMemberPostingRole(value as SourceMemberRole)
-                }
-              />
-            </div>
-          )}
+          <div className="flex flex-col justify-start">
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:194]: Don't think this adds anything.

```suggestion
          <div className="flex flex-col">
```

File: packages/shared/src/components/squads/Details.tsx [hunk 3]
```diff
@@ -191,24 +191,22 @@ export function SquadDetails({
...
+            <h4 className="mb-2 font-bold typo-headline text-theme-label-primary">
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:195]: This should be enough:
```suggestion
            <h4 className="mb-2 typo-headline">
```

File: packages/shared/src/components/squads/utils.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { PromptOptions } from '../../hooks/usePrompt';
 import { SquadForm } from '../../graphql/squads';
 
 export enum ModalState {
-  Details = 'Squad details',
```
Comment (rebelchris) [inline packages/shared/src/components/squads/utils.tsx:7]: Is this actually still used? Not quite sure we use this flow anymore.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Some minor comments, but otherwise happy.

File: packages/shared/src/components/squads/utils.tsx (hunk 4 above)
Comment (capJavert) [reply packages/shared/src/components/squads/utils.tsx:7]: It is, but below it `modalStateToScreenValue`, left it because I did not know the context.

Comment (rebelchris) [reply packages/shared/src/components/squads/utils.tsx:7]: Ah let's leave it might be needed for typesafety or something

**Review by rebelchris — APPROVED** [review]
Amazing, nice work 🙏

**Pull request merged**